guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/DeleteProjectObserverBridge.java

```diff
@@ -34,20 +35,33 @@
 import org.uberfire.workbench.events.ResourceDeleted;
 import org.uberfire.workbench.events.ResourceDeletedEvent;
 
+/**
+ * A bridge between changes made to an underlying VFS and Project abstractions. Projects can be deleted
+ * either through the Workbench's UI or via REST or on a cloned repository that's pushed back to the Workbench.
+ * Interested observers are signaled of a Project deletion event when a pom.xml file is detected as deleted.
+ */
 @ApplicationScoped
-public class PathObserverService {
+public class DeleteProjectObserverBridge {
```

Comment:
A lot better name. Good to have a class for single observation task, delete in this case.